### ISSUE TITLE
Improve const-correctness

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#242]
**Reported by elfring on 18 Mar 37431241 06:40 UTC**
Would you like to apply the advices from a [well-known article](http://en.wikipedia.org/wiki/Const-correctness) to more places in your source files?[suggest to add the key word "const" to the type specifiers for parameters like the following.[[BR]([BR]]
I)]
- string (functions "[txfRenderString](source:/trunk/boinc/api/texfont.h@9956#L89)", "txfRenderFancyString")[name (function "[source:/trunk/boinc/api/texture.h@6680#L14 read_tga_texture]([BR]]
  -)")[dir (function "[source:/trunk/boinc/api/txf_util.h@9838#L30 txf_load_fonts]([BR]]
  -)")[s (function "[source:/trunk/boinc/api/txf_util.h@9838#L31 txf_render_string]([BR]]
  -)")[N, label ([source:/trunk/boinc/client/stream.C@9785#L74 global variables]([BR]]
  -))

Migrated-From: http://boinc.berkeley.edu/trac/ticket/243

**Comment by romw** [comment]
**Commented by Nicolas on 11 Jan 37751963 16:53 UTC**
Removing component, since this could apply to the whole BOINC codebase.

**Comment by romw** [comment]
**Commented by elfring on 7 Oct 37797663 13:20 UTC**
Another example:[icon "usage.xpm" for the client GUI was tagged as read-only already. Would you also like to apply this property to more pictures to avoid messages like the following?[[BR]([BR]]
The)]

res/warning.xpm:63: Warnung: deprecated conversion from string constant to char*

**Comment by romw** [comment]
**Commented by elfring on 10 Dec 38489614 09:46 UTC**
Would you like to reduce more warnings that I get during compilation by GCC 4.3.1?

**Comment by elfring** [comment]
How are the chances to [improve the involved application programming interfaces](https://github.com/BOINC/boinc/blob/5858534492902c1e215a74eda15b0b734109e12a/api/boinc_api.h#L115 "Update candidate")?

**Comment by ChristianBeer** [comment]
A lot has been done to fix this since this was reported. For new ones please open a new issue.

**Comment by elfring** [comment]
Is [the function “setMacIcon” still relevant](https://github.com/BOINC/boinc/blob/f0c39bdf5117d8f7dd5092033971d7f700bd22dc/api/boinc_api.h#L118 "Remaining update candidates?") as an update candidate for this issue?

**Comment by CharlieFenton** [comment]
Since setMacIcon has been deprecated, it is no longer relevant.

**Comment by CharlieFenton** [comment]
When I wrote "Since setMacIcon has been deprecated, it is no longer relevant." I should have written "obsolete and removed from the Xcode project" rather than just "deprecated." See my GIT commits 35ec133 (2/27/17) and 7e213cf 2/28/17). 

The comments I added to mac_icon.cpp and make_app_icon_h.cpp describe a better way to add icons to project executable files that is the currently recommended method and does not use the long deprecated resource fork. I have not removed these files from the GIT repository so that the information in these comments remains available.
